pkg/deploy/api/validation/validation.go

```diff
@@ -37,7 +37,7 @@ func ValidateDeploymentConfig(config *deployapi.DeploymentConfig) fielderrors.Va
 	if config.Spec.Replicas < 0 {
 		allErrs = append(allErrs, fielderrors.NewFieldInvalid("spec.replicas", config.Spec.Replicas, "replicas cannot be negative"))
 	}
-	if config.Spec.Selector == nil || len(config.Spec.Selector) == 0 {
+	if len(config.Spec.Selector) == 0 {
```

Comment:
> Seems like you ought to be able to do some validation on the correctness of the selector/template.labels combination, right?

I don't think this is necessary. If the selector is specified then we don't really care about the template labels. If the dc selector stays empty and there are template labels, then we should use those as the selector. If in the last case, both are empty then it;s a user error and validation should have a say.